File: interface/src/main/java/eu/nyerel/panda/monitoringresult/calltree/AggregatedCallTreeNode.java
package eu.nyerel.panda.monitoringresult.calltree;

import java.util.ArrayList;

/**
 * @author Rastislav Papp (rastislav.papp@gmail.com)
 */
public class AggregatedCallTreeNode extends CallTreeNode {

    private int aggregateCount = 1;

    public AggregatedCallTreeNode() {
    }

    public AggregatedCallTreeNode(CallTreeNode node) {
        setId(node.getId());
        setType(node.getType());
        setDescription(node.getDescription());
        setDuration(node.getDuration());
        setParent(node.getParent());
        setChildren(new ArrayList<CallTreeNode>());

        for (CallTreeNode child : node.getChildren()) {
            getChildren().add(new AggregatedCallTreeNode(child));
        }
    }

    public int getAggregateCount() {
        return aggregateCount;
    }

    public void setAggregateCount(int aggregateCount) {
        this.aggregateCount = aggregateCount;
    }

}

File: panda-ij-plugin/src/eu/nyerel/panda/ijplugin/model/AggregatedCallTreeNode.kt
package eu.nyerel.panda.ijplugin.model

import eu.nyerel.panda.monitoringresult.calltree.CallTreeNode
import eu.nyerel.panda.monitoringresult.calltree.CallTreeNodeDuration
import eu.nyerel.panda.monitoringresult.calltree.CallTreeNodeType

/**
 * @author Rastislav Papp (rastislav.papp@gmail.com)
 */
data class AggregatedCallTreeNode(private val node: CallTreeNode) : CallTreeNode {

    override val id: Long
        get() = node.id
    override val type: CallTreeNodeType
        get() = node.type
    override var duration: CallTreeNodeDuration = node.duration
    override val description: String
        get() = node.description
    override var parent: CallTreeNode? = null
        get() = node.parent
    override var children: List<CallTreeNode> = ArrayList()
    var aggregateCount = 1

}
